erpnext/accounts/doctype/process_deferred_accounting/process_deferred_accounting.js

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2019, Frappe Technologies Pvt. Ltd. and contributors
+// For license information, please see license.txt
+
+frappe.ui.form.on('Process Deferred Accounting', {
+	setup: function(frm) {
+		frm.set_query("document_type", function() {
+			return {
+				filters: {
+					'name': ['in', ['Sales Invoice', 'Purchase Invoice']]
+				}
+			};
+		});
+	},
+
+	before_submit: function() {
+		frappe.db.get_single_value('Accounts Settings', 'automatically_process_deferred_accounting_entry')
+			.then(value => {
+				if(value) {
+					frappe.throw(__('Manual entry cannot be created! Disable automatic entry for deferred accounting in accounts settings and try again'));
+				}
+			});
+	},
+
+	end_date: function(frm) {
```

Comment:
Move this validation to server side inside validate trigger

Reply:
added server side validation too